src/playermat/Playmat.ttslua

```diff
@@ -457,133 +434,102 @@ function replenishTokens(card, count, replenish)
-end
-
--- contains position and amount of boxes for the upgradesheets that change uses
--- Alchemical Distillation, Damning Testimony, Living Ink and Hyperphysical Shotcaster
-local customizationsTable = {
-  ["09040"] = {5, 2},
-  ["09059"] = {2, 2},
-  ["09079"] = {3, 2},
-  ["09119"] = {6, 4}
-}
-
-function spawnTokensFor(object)
-  local cardMetadata = JSON.decode(object.getGMNotes()) or {}
-  local type = nil
-  local token = nil
-  local tokenCount = 0
-  if cardMetadata.uses ~= nil then
-    -- Uses structure underwent a breaking change in 2.4.0, have to check to see if this is
-    -- a single entry or an array.  This is ugly and duplicated, but impossible to replicate the
-    -- multi-spawn vs. single spawn otherwise.  TODO: Clean this up when 2.4.0 has been out long
-    -- enough that saved decks don't have old data
-    if cardMetadata.uses.count != nil then
-      type = cardMetadata.uses.type
-      token = cardMetadata.uses.token
-      tokenCount = cardMetadata.uses.count
-      if activeInvestigatorId == "03004" and type == "Charge" then tokenCount = tokenCount + 1 end
-
-      log("Spawning tokens for "..object.getName()..'['..object.getDescription()..']: '..tokenCount.."x "..token)
-      spawnTokenGroup(object, token, tokenCount)
-    else
-      for i, useInfo in ipairs(cardMetadata.uses) do
-        type = useInfo.type
-        token = useInfo.token
-        tokenCount = useInfo.count
-
-        -- additional uses for certain customizable cards (by checking the upgradesheets)
-        if customizationsTable[cardMetadata.id] ~= nil then
-          for _, obj in ipairs(searchArea(PLAY_ZONE_POSITION, PLAY_ZONE_SCALE)) do
-            local obj = obj.hit_object
-            if obj.tag == "Card" then
-              local notes = JSON.decode(obj.getGMNotes()) or {}
-              if notes.id == (cardMetadata.id .. "-c") then
-                local pos = customizationsTable[cardMetadata.id][1]
-                local boxes = customizationsTable[cardMetadata.id][2]
-                if obj.getVar("markedBoxes")[pos] == boxes then tokenCount = tokenCount + 2 end
-                break
-              end
-            end
+function syncCustomizableMetadata(card)
+  local cardMetadata = JSON.decode(card.getGMNotes()) or { }
+  if cardMetadata ~= nil and cardMetadata.customizations ~= nil then
+    for _, obj in ipairs(searchArea(PLAY_ZONE_POSITION, PLAY_ZONE_SCALE)) do
+      local obj = obj.hit_object
+      local notes = JSON.decode(obj.getGMNotes()) or { }
+      if notes.id == (cardMetadata.id .. "-c") then
+        for i, customization in ipairs(cardMetadata.customizations) do
+          if obj.getVar("markedBoxes")[i] == customization.xp
+              and customization.replaces ~= nil
+              and customization.replaces.uses ~= nil then
+            cardMetadata.uses = customization.replaces.uses
+            card.setGMNotes(JSON.encode(cardMetadata))
```

Comment:
How does this work with cards like `Refine` that add checkboxes during play?

Reply:
At the moment it doesn't.  We need to add something to the Upgrade Sheets to push a call to this for all cards on the mat.  I think we can do that in a followon.

But I also don't think it affects anything.  For everything except Runic Axe, the only changes are to the initial number of uses.  So a change won't affect anything already in play.  The only issue would be Runic Axe's extra replenishment, which is still handled explicitly.